File: Python/Program/Car.py
# Saya Syifa Azzahra NIM 2207308 mengerjakan soal Latihan 4 dalam mata kuliah 
# Desain Pemrograman Berorientasi Objek untuk keberkahan-Nya maka saya tidak melakukan kecurangan seperti yang telah
# dispesifikasikan. Aamiin. 

from Vehicle import Vehicle

class Car(Vehicle):
# kelas yang digunakan untuk mengimplementasikan kelas Car
    
    def __init__(self, platNomor, merk, tahunProduksi, warna, jenisMobil, jumlahKursi, jumlahPintu, namaVehicleStorage, lokasi):
        super().__init__(platNomor, merk, tahunProduksi, warna)
        self.jenisMobil = jenisMobil
        self.jumlahKursi = jumlahKursi
        self.jumlahPintu = jumlahPintu
        self.namaVehicleStorage = namaVehicleStorage
        self.lokasi = lokasi

    # mengeset dan mengembalikan atribut-atribut pada kelas Car
    def setJenisMobil(self, jenisMobil):
        self.jenisMobil = jenisMobil
    
    def getJenisMobil(self):
        return self.jenisMobil

    def setJumlahKursi(self, jumlahKursi):
        self.jumlahKursi = jumlahKursi

    def getJumlahKursi(self):
        return self.jumlahKursi
    
    def setJumlahPintu(self, jumlahPintu):
        self.jumlahPintu = jumlahPintu

    def getJumlahPintu(self):
        return self.jumlahPintu

    def setNamaVehicleStorage(self, namaVehicleStorage):
        self.namaVehicleStorage = namaVehicleStorage

    def getNamaVehicleStorage(self):
        return self.namaVehicleStorage
    
    def setLokasi(self, lokasi):
        self.lokasi = lokasi

    def getLokasi(self):
        return self.lokasi

File: CPP/Program/Car.cpp
/* Saya Syifa Azzahra NIM 2207308 mengerjakan soal Latihan 4 dalam mata kuliah 
Desain Pemrograman Berorientasi Objek untuk keberkahan-Nya maka saya tidak melakukan kecurangan seperti yang telah
dispesifikasikan. Aamiin. */

#pragma once

#include <string>
#include <iostream>
#include <vector>
#include "Vehicle.cpp"

using namespace std;

class Car : public Vehicle 
{
    private:
        string jenisMobil;
        string jumlahKursi;
        string jumlahPintu;

    public:

        // Konstruktor
        Car(string platNomor, string merk, string tahunProduksi, string warna, string jenisMobil, string jumlahKursi, string jumlahPintu)
            : Vehicle(platNomor, merk, tahunProduksi, warna), jenisMobil(jenisMobil), jumlahKursi(jumlahKursi), jumlahPintu(jumlahPintu) 
            {
                this->jenisMobil = jenisMobil;
                this->jumlahKursi = jumlahKursi;
                this->jumlahPintu = jumlahPintu;
            }

        // Getter dan Setter untuk JenisMobil
        void setJenisMobil(string jenisMobil) 
        {
            this->jenisMobil = jenisMobil;
        }

        string getJenisMobil() 
        {
            return this->jenisMobil;
        }

        // Getter dan Setter untuk JumlahKursi
        void setJumlahKursi(string jumlahKursi) 
        {
            this->jumlahKursi = jumlahKursi;
        }

        string getJumlahKursi() 
        {
            return this->jumlahKursi;
        }

        // Getter dan Setter untuk JumlahPintu
        void setJumlahPintu(string jumlahPintu) 
        {
            this->jumlahPintu = jumlahPintu;
        }

        string getJumlahPintu() 
        {
            return this->jumlahPintu;
        }

        ~Car()
        {

        }
};